arbnode/sequencer.go

```diff
@@ -362,6 +364,31 @@ func (s *Sequencer) preTxFilter(_ *params.ChainConfig, header *types.Header, sta
 			return err
 		}
 	}
+	if options != nil && len(options.KnownAccounts) > 0 {
+		for address, rootHashOrSlots := range options.KnownAccounts {
+			if rootHashOrSlots.RootHash != nil {
+				trie := statedb.StorageTrie(address)
+				if trie == nil {
+					// TODO
+					return errors.New("Trie not found for address")
+				}
+				if trie.Hash() != *rootHashOrSlots.RootHash {
+					// TODO
+					return errors.New("Rejected")
+				}
+			} else if len(rootHashOrSlots.SlotValue) > 0 {
+				for slot, value := range rootHashOrSlots.SlotValue {
+					if statedb.GetState(address, slot) != value {
+						// TODO
+						return errors.New("Rejected")
+					}
+				}
+			} else {
+				// TODO
+				return errors.New("Invalid")
```

Comment:
I think we should accept the tx in this case, e.g. if an account was specified as an empty map.